### ISSUE TITLE
Crouton chroot xenial unity issue when installing after months of trying. PLEASE HELP ASAP!

### ISSUE THREAD

**Issue by altea1566** [dnschneid/crouton#4936]
```
Please paste the output of the following command here: sudo edit-chroot -all
```
name: xenial
encrypted: no
Entering /mnt/stateful_partition/crouton/chroots/xenial...
crouton: version 1-20221126232257~master:ae594340
release: xenial
architecture: amd64
targets: unity
host: version 15357.0.0 (Official Build) dev-channel octopus 
kernel: Linux localhost 4.14.254-20285-g8f8e6ec4f88c #1 SMP PREEMPT Tue Feb 21 04:52:11 PST 2023 x86_64 x86_64 x86_64 GNU/Linux
freon: yes
Unmounting /mnt/stateful_partition/crouton/chroots/xenial...

#### Please describe your issue:

Its been just over a month now and I've been having this problem:

I use sudo sh ~/Downloads/crouton -t unity it says sh: 0: Refusing to exec /home/chronos/user/Downloads/crouton from noexec mount; see https://chromium.googlesource.com/chromiumos/docs/+/master/security/noexec_shell_scripts.md

So then I went to this [Reddit Page](https://www.reddit.com/r/Crouton/comments/gj2cy8/getting_refusing_to_exec/) and it says to use the commands sudo cp ~/Downloads/crouton /usr/local and then sudo sh /usr/local/crouton -t unity, which worked!

But then at the end, it gave me this: 

sed: can't read common/cras_messages.h: No such file or directory
Failed to complete chroot setup.
The chroot setup script may be broken. Your chroot is not fully configured.
Removing the chroot setup script. You may want to update your chroot again.
UID 1000 not found in xenial
Unmounting /mnt/stateful_partition/crouton/chroots/xenial...

Then I reported this on reddit and they said to change the chroot location or something to sudo install -Dt /usr/local/bin -m 755 ~/Downloads/crouton

Which worked again! But once again, at the end, it gave me: 

sed: can't read common/cras_messages.h: No such file or directory
Failed to complete chroot setup.
Unmounting /mnt/stateful_partition/crouton/chroots/xenial...

Btw, I'm using 112.0.5599.0 dev chrome version build (latest) and my chromebook model is a 2022 Lenovo Chromebook gemini lake 64 bit 

And also, I've powerwashed and switched back to non developer mode and back twice now and same thing. It used to work like 8 months ago but then I reset my chromebook.

Please help because Ive done so much research, I've covered every thing on the internet related to this problem and tried every way.

I really appreaciate your time and hope I didn't waste your time with this long complain

Thanks.

**Comment by altea1566** [comment]
Just letting you know, when doing this to install kali linux on my chromebook, this happened

sed: can't read common/cras_messages.h: No such file or directory
Failed to complete chroot setup.
The chroot setup script may be broken. Your chroot is not fully configured.
Removing the chroot setup script. You may want to update your chroot again.
UID 1000 not found in kali
Unmounting /mnt/stateful_partition/crouton/chroots/kali...

Basically the same thing, please help asap though.

**Comment by reboun** [comment]
I am having the same problem

**Comment by st106931** [comment]
Same issue over here

**Comment by lorenkjr1025** [comment]
I am having the same problem too. I can not install linux with crouton on my at all, as it will not read messages.  What can be done to fix this?

**Comment by NateTheClickbait** [comment]
This Thread has new code for it but I dont know how to inject that code into crouton[ ](https://github.com/dnschneid/crouton/issues/4946)

**Comment by OrangeGlowest** [comment]
How would I enter the chroot though

**Comment by OplopanaxHorridus** [comment]
> Not sure which branch is stable. But it worked for me to switch to crouton-silence branch, download the .zip of everything, and run crouton with `installer/main.sh` in the directory where it is copied. This is the crouton-silence branch: https://github.com/dnschneid/crouton/tree/silence Now I don't get the error - which was happening no matter what options for install.
> 
> I copied the unzipped folder of crouton-silence to /usr/local/ as per the README suggestion for when you are editing crouton - although I made no edits to the downloaded code.
> 
> Some example code which is working now: `sudo bash /usr/local/crouton-silence/installer/main.sh -t kde-desktop -r xenial`

I tried this but I got bash: usr/local/crouton-silence/installer/main.sh -t kde-desktop -r xenial: no such file or directory.

**Comment by Apocalypse10110** [comment]
I have the same issue
